stdlib/public/core/UnsafeRawBufferPointer.swift.gyb

```diff
@@ -115,8 +115,23 @@ public struct Unsafe${Mutable}RawBufferPointer {
     _debugPrecondition(count >= 0, "${Self} with negative count")
     _debugPrecondition(count == 0 || start != nil,
       "${Self} has a nil start and nonzero count")
+
+    self.init(_uncheckedStart: start, count: _assumeNonNegative(count))
+  }
+
+  @_alwaysEmitIntoClient
+  internal init(
+    _uncheckedStart start: Unsafe${Mutable}RawPointer?,
+    count: Int
+  ) {
+    _internalInvariant(count >= 0, "${Self} with negative count")
+    _internalInvariant(
+      count == 0 || start != nil,
+      "${Self} has a nil start and nonzero count"
+    )
+
     _position = start
-    _end = start.map { $0 + _assumeNonNegative(count) }
+    _end = start.map { $0 + count }
```

Comment:
We may want to still invoke the `Builtin.assumeNonNegative_` intrinsic to get to full warp speed. (Then again, how closely do we want to emulate the mistakes of C/C++? 🤔)

Reply:
it's still included in the public initializer:
```swift
self.init(_uncheckedStart: start, count: _assumeNonNegative(count))
```

Reply:
I just went and added it back to both the checked and unchecked 👍